### PR TITLE
pass bytes to save_xform

### PR DESCRIPTION
**Pull request by nickpell**
https://sentry.io/dimagi/commcarehq/issues/793334137/
and
https://sentry.io/dimagi/commcarehq/issues/793392230/

### CODE REVIEW

**Review by gbova — APPROVED** [review]

File: corehq/apps/app_manager/views/forms.py [hunk 1]
```diff
@@ -490,13 +490,13 @@ def patch_xform(request, domain, app_id, form_unique_id):
     current_xml = form.source
     dmp = diff_match_patch()
     xml, _ = dmp.patch_apply(dmp.patch_fromText(patch), current_xml)
-    xml = save_xform(app, form, xml)
+    xml = save_xform(app, form, xml.encode('utf-8'))
```
Comment (millerdev) [inline corehq/apps/app_manager/views/forms.py]: After this assignment `xml` apparently has a different type (bytes instead of text) than it did before, which is a little hard to follow. I should note that this is an inferrence based on how you've changed the code in this PR.

Comment (nickpell) [reply corehq/apps/app_manager/views/forms.py]: I agree - bytes/text is a bit of a mess right now in app_manager.  Just getting form.source to have a text interface turned out to be a lot more complicated than I expected.  I'm going to hold off on making additional changes until I have a passing build in python 3 to test against (we don't have proper unicode tests in this area for python 2).

Definitely long-term we want to have a proper unicode sandwich in this area of the code.  In terms of planning, I'd like to do cleanup like this once we have python 3 running on staging and are in a holding period before deploying to production.
